### PR TITLE
Site: Add guidelines for JIRA's fix version field

### PR DESCRIPTION
**Pull request by zabetak**
Add guidelines for setting correctly the fix version field in JIRA, as proposed by Francis Chuang in a discussion on the dev list.

### CODE REVIEW

File: site/develop/index.md [hunk 1]
```diff
@@ -67,6 +67,26 @@ helps to agree on the general approach. Log a
 [JIRA case](https://issues.apache.org/jira/browse/CALCITE) for your
 proposed feature or start a discussion on the dev list.
 
+Before opening up a new JIRA case, have a look in the existing issues.
+The feature or bug that you plan to work on may already be there.  
+
+If you are going to take on the issue right away assign it to your self.
```
Comment (julianhyde) [inline site/develop/index.md]: s/your self/yourself/

File: site/develop/index.md [hunk 2]
```diff
@@ -67,6 +67,26 @@ helps to agree on the general approach. Log a
...
+To assign issues to your self you have to be registered as a contributor 
+to the project. In order to do that, send an email to the developers list
```
Comment (julianhyde) [inline site/develop/index.md]: s/registered as a contributor to the project/registered in JIRA as a contributor/

(I don't want people to think that 'contributor' is an Apache designation.)

File: site/develop/index.md [hunk 3]
```diff
@@ -67,6 +67,26 @@ helps to agree on the general approach. Log a
...
+and provide your JIRA username.   
+
+If you think the issue can make it to the ongoing release set the fix version
```
Comment (julianhyde) [inline site/develop/index.md]: "If you are committed to fixing the issue before the upcoming release"

File: site/develop/index.md [hunk 4]
```diff
@@ -67,6 +67,26 @@ helps to agree on the general approach. Log a
...
+accordingly (e.g., 1.20.0), otherwise leave it as blank.
+
+If you pick up an existing issue and start working on it, set the fix
```
Comment (julianhyde) [inline site/develop/index.md]: I'd strike this paragraph. If you pick up an existing issue, mark it 'in progress', and when it's finished flag 'pull-request-available'. You never know whether you will finish in time, so there's no point making promises.

Frankly, it's pretty rare that anyone besides the release manager would set fixVersion. Sometimes if pull-request-available is set, a reviewer will set fixVersion to remind them to review before the release.

File: site/develop/index.md (hunk 1 above)
Comment (zabetak) [reply site/develop/index.md]: Done

File: site/develop/index.md (hunk 2 above)
Comment (zabetak) [reply site/develop/index.md]: Done

File: site/develop/index.md (hunk 4 above)
Comment (zabetak) [reply site/develop/index.md]: Agreed.

File: site/develop/index.md (hunk 3 above)
Comment (zabetak) [reply site/develop/index.md]: Done

**Pull request merged**